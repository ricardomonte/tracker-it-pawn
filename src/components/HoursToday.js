import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgressbar } from 'react-circular-progressbar';
import 'react-circular-progressbar/dist/styles.css';

const HoursToday = ({ myHours, weeklyPlayTime }) => {
  const d = new Date().getMonth() + 1;
  const filteredDate = myHours.filter((element) => {
    const month = element.play.split('-')[1];
    return (d === +month);
  });

  const total = filteredDate.reduce((total, element) => (total + element.hour), 0);

  const percentage = (100 * total) / (weeklyPlayTime * 4);
  const percentRound = Math.round(percentage);

  return (
    <div>
      <h3>Hours played this Month (approx)</h3>
      <CircularProgressbar
        value={percentRound}
        text={`${percentRound} %`}
        maxValue={weeklyPlayTime * 4}
      />
    </div>
  );
};

HoursToday.propTypes = {
  weeklyPlayTime: PropTypes.number.isRequired,
  myHours: PropTypes.instanceOf(Array).isRequired,
};

export default HoursToday;
