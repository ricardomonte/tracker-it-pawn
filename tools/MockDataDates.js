const MockDataDates = [
  {
    id: 1,
    name: 'test data 1',
    cover: '//mock.data.com/test.jpg',
    play: '2021-5-21',
    rating: 72,
  },
  {
    id: 2,
    name: 'test data 2',
    cover: '//mock.data.com/test2.jpg',
    play: '2021-5-21',
    rating: 80,
  },
  {
    id: 3,
    name: 'test data 3',
    cover: '//mock.data.com/test3.jpg',
    play: '2021-5-21',
    rating: 10,
  },
  {
    id: 4,
    name: 'test data 4',
    cover: '//mock.data.com/test4.jpg',
    play: '2021-5-21',
    rating: 95.67,
  },
  {
    id: 5,
    name: 'test data 5',
    cover: '//mock.data.com/test5.jpg',
    play: '2021-5-21',
    rating: 100,
  },
];

export default MockDataDates;