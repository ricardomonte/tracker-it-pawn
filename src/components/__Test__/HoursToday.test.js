import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import MockDataDates from '../../../tools/MockDataDates';
import HoursToday from '../HoursToday';

test('Content from Expense matches snapshot', () => {
  const content = render(
    <HoursToday myHours={MockDataDates} weeklyPlayTime={15} />
  );
  expect(content.container).toMatchSnapshot();
});