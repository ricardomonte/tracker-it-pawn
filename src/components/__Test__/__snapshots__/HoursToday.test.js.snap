// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content from HoursToday matches snapshot 1`] = `
<div>
  <div>
    <h3>
      Hours played this Month (approx)
    </h3>
    <svg
      class="CircularProgressbar "
      data-test-id="CircularProgressbar"
      viewBox="0 0 100 100"
    >
      <path
        class="CircularProgressbar-trail"
        d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
        fill-opacity="0"
        stroke-width="8"
        style="stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 0px;"
      />
      <path
        class="CircularProgressbar-path"
        d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
        fill-opacity="0"
        stroke-width="8"
        style="stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: NaNpx;"
      />
      <text
        class="CircularProgressbar-text"
        x="50"
        y="50"
      >
        NaN %
      </text>
    </svg>
  </div>
</div>
`;
